ratis-server/src/main/java/org/apache/ratis/server/raftlog/segmented/SegmentedRaftLogCache.java

```diff
@@ -151,8 +151,8 @@ long maxEndIndex() {
       if (toTruncate != null) {
         max = toTruncate.endIndex;
       }
-      for(SegmentFileInfo d : toDelete) {
-        max = Math.max(max, d.endIndex);
+      if (toDelete != null && toDelete.length > 0) {
+        max = Math.max(max, toDelete[0].endIndex);
```

Comment:
Let's keep the origin code since it is more robust -- it does not depend on the ordering.

Reply:
I have reverted this line